util/src/snowflake.rs

```diff
@@ -0,0 +1,158 @@
+//! A module providing the Snowflake trait for defining extractable information from a Discord
```

Comment:
Let's drop the "a module", and say something like "Snowflake trait for defining.."